code/datums/components/kneecapping.dm

```diff
@@ -0,0 +1,90 @@
+ *
+ * Kneecapping requires the target to either be on the floor, immobilised or buckled to something. And also to have an appropriate leg.
+ *
+ * Passing all the checks will cancel the entire attack chain.
+ */
+/datum/element/kneecapping
+	element_flags = ELEMENT_DETACH
+
+/datum/element/kneecapping/Attach(datum/target)
+	if(!isitem(target))
+		stack_trace("Kneecapping element added to non-item object: \[[target]\]")
+		return ELEMENT_INCOMPATIBLE
+
+	var/obj/item/target_item = target
+
+	if(target_item.force < WOUND_MINIMUM_DAMAGE)
+		stack_trace("Kneecapping element added to item with too little force to wound: \[[target]\]")
+		return ELEMENT_INCOMPATIBLE
+
+	. = ..()
+
+	if(. == ELEMENT_INCOMPATIBLE)
+		return
+
+	RegisterSignal(target, COMSIG_ITEM_ATTACK_SECONDARY , .proc/try_kneecap_target)
+
+/datum/element/kneecapping/Detach(datum/target)
+	UnregisterSignal(target, COMSIG_ITEM_ATTACK_SECONDARY)
+
+	return ..()
+
+/**
+ * Signal handler for COMSIG_ITEM_ATTACK_SECONDARY. Does checks for pacifism, zones and target state before either returning nothing
+ * if the special attack could not be attempted, performing the ordinary attack procs instead - Or cancelling the attack chain if
+ * the attack can be started.
+ */
+/datum/element/kneecapping/proc/try_kneecap_target(obj/item/source, mob/living/carbon/target, mob/attacker, params)
+	SIGNAL_HANDLER
+
+	if((attacker.zone_selected != BODY_ZONE_L_LEG) && (attacker.zone_selected != BODY_ZONE_R_LEG))
+		return
+
+	if(HAS_TRAIT(attacker, TRAIT_PACIFISM))
+		return
+
+	if(!iscarbon(target))
+		return
+
+	if(!target.buckled && !HAS_TRAIT(target, TRAIT_FLOORED) && !HAS_TRAIT(target, TRAIT_IMMOBILIZED))
+		return
+
+	var/obj/item/bodypart/leg = target.get_bodypart(attacker.zone_selected)
+
+	if(!leg)
+		return
+
+	. = COMPONENT_SECONDARY_CANCEL_ATTACK_CHAIN
+
+	INVOKE_ASYNC(src, .proc/do_kneecap_target, source, leg, target, attacker)
+
+/**
+ * After a short do_mob, attacker applies damage to the given leg with a significant wounding bonus, applying the weapon's force as damage.
+ */
+/datum/element/kneecapping/proc/do_kneecap_target(obj/item/weapon, obj/item/bodypart/leg, mob/living/carbon/target, mob/attacker)
+	if(LAZYACCESS(attacker.do_afters, weapon))
+		return
+
+	attacker.visible_message(span_warning("[attacker] carefully aims [attacker.p_their()] [weapon] for a swing at [target]'s kneecaps!"), span_danger("You carefully aim \the [weapon] for a swing at [target]'s kneecaps!"))
+	log_combat(attacker, target, "started aiming a swing to break the kneecaps of", weapon)
+
+	if(do_mob(attacker, target, 3 SECONDS, interaction_key = weapon))
+		attacker.visible_message(span_warning("[attacker] swings [attacker.p_their()] [weapon] at [target]'s kneecaps!"), span_danger("You swing \the [weapon] at [target]'s kneecaps!"))
+		var/datum/wound/blunt/severe/severe_wound_type = /datum/wound/blunt/severe
+		var/datum/wound/blunt/critical/critical_wound_type = /datum/wound/blunt/critical
```

Comment:
what if this element has been applied to a non-blunt weapon?

Reply:
well fuck the person that did it then

Reply:
I just picked blunt as the baseline for wounding interactions out of my ass.

Even though the element can technically be added to any item with enough force, from var edited plastic spoons to spears, the behaviour can be further enhanced later if someone wanted to add checks for the injury thresholds for piercing and slashing wounds too.

Right now the element is only added to a single blunt weapon outside of admin abuse. I wanted to keep the option open for admins to add it to any weapon even if it doesn't use equivalent threasholds for that weapon's wounding type.